modules/packages/BLAS.chpl

```diff
@@ -695,8 +695,662 @@ module BLAS {
+      cblas_zdotu_sub (N, X, incX, Y, incY, res);
+    }
+    otherwise {
+        compilerError("Unknown type in dotu_sub");
+    }
+  }
+
+  return res;
+
+}
+
+
+/*
+    Wrapper for `DOTC_SUB routines <http://www.netlib.org/lapack/explore-html/da/df6/group__complex__blas__level1_gadd72f1b633553acc77250e32bc704a78.html#gadd72f1b633553acc77250e32bc704a78>`_
+
+    Obtains the dot product of conjugated X vector with Y vector::
+
+       dotc = conj(X)*Y
+
+    Input:
+
+    X: Conjugated complex vector with N elements.
+    Y: Complex vector with N elements.
+    incX: defines the increment for the vector X.
+    incY: defines the increment for the vector Y.
+
+    Return:
+
+    dotc:  Has the result of complex dot product.
+
+
+*/
+proc dotc(X: [?D]?eltType, Y: [D]eltType, incY: c_int = 1, incX: c_int = 1)
+ where D.rank == 1 {
+
+  const N = D.size: c_int;
+
+  var res: eltType;
+
+  select eltType {
+    when complex(64) do{
+      cblas_cdotc_sub (N, X, incX, Y, incY, res);
+    }
+    when complex(128) do{
+      cblas_zdotc_sub (N, X, incX, Y, incY, res);
+    }
+    otherwise {
+        compilerError("Unknown type in dotc_sub");
+    }
+  }
+  return res;
+}
+
+/*
+    Wrapper for `DSDOT routines <http://www.netlib.org/lapack/explore-html/df/d28/group__single__blas__level1_gaddc89585ced76065053abffb322c5a22.html#gaddc89585ced76065053abffb322c5a22>`_
+
+    Returns the dot product of two real(32) vectors as a real(64), using
+    real(64) precision internally::
+
+       d = X*Y
+
+    Input:
+
+    Y,X:Vector with N elements.
+    incX: defines the increment for the vector X.
+    incY: defines the increment for the vector Y.
+
+    Return:
+
+    Scalar value of dot product.
+
+
+*/
+proc dsdot(X: [?D] real(32), Y: [D] real(32), incY: c_int = 1,incX: c_int = 1): real(64)
+ where D.rank == 1 {
+
+  const N = D.size: c_int;
+  return cblas_dsdot (N, X, incX, Y, incY);
+}
+
+
+/*
+    Wrapper for `SDSDOT routines <http://www.netlib.org/lapack/explore-html/df/d28/group__single__blas__level1_gaddc89585ced76065053abffb322c5a22.html#gaddc89585ced76065053abffb322c5a22>`_
+
+    Returns the dot product of two real(32) vectors as a real(32), using
+    real(64) precision internally::
+
+       d = X*Y
+
+    Input:
+
+    Y,X:Vector with N elements.
+    incX: defines the increment for the vector X.
+    incY: defines the increment for the vector Y.
+
+    Return:
+
+    Scalar value of dot product.
+
+
+*/
+proc sdsdot(X: [?D] real(32), Y: [D] real(32), incY: c_int = 1,incX: c_int = 1): real(32)
+ where D.rank == 1 {
+
+  var alpha = 0: real(32);
+  const N = D.size: c_int;
+
+  return cblas_sdsdot (N, alpha, X, incX, Y, incY);
+}
+
+/*
+    Wrapper for `NRM2 routines <http://www.netlib.org/lapack/explore-html/df/d28/group__single__blas__level1_ga35c2ec0e9bfdaa545320c2a134fcc471.html#ga35c2ec0e9bfdaa545320c2a134fcc471>`_
+
+    Returns the  Euclidian norm of vector X::
+
+      d = ||X||
+
+    Input:
+
+    X:Vector with N elements.
```

Comment:
For consistency, I think it would be good to decide on either one or zero spaces after a colon in these documentation snippets.